### ISSUE TITLE
shutdown of application instance deregisters entire application

### ISSUE THREAD

**Issue by xf57212** [Netflix/eureka#1230]
Spring Boot version: 2.0.4.RELEASE
Spring Cloud version: Finchley.RELEASE
eureka-client version: 1.9.2

We have 60+ instances of a microservice running in a docker swarm. We scale these microservices dynamically based on need, so the number of instances are constantly in a state of flux, going up and down.

Our problem is when a single instance is shut down due to lack of need, it deregisters the entire application from the registry instead of just the instance. This causes us to drop data because the registry thinks the service no longer exists until another instance somehow figures out it needs to reregister a few seconds later.

I have tried setting the config option `shouldUnregisterOnShutdown: false`, which does indeed prevent the service from deregistering. But now if a single instance is shut down, it registers the service in a "DOWN" state. Even though there are still at least 59 instances running.

While in a down state, the registry prevents anything from talking to the service.

How can I deregister a single instance of the application, NOT put the application in a down state, and continue processing seemlessly with the remaining instances that are still running?

**Comment by troshko111** [comment]
This should not be happening in any case (graceful instance deregister or abrupt heartbeat stop), the other instances will stay as UP. If you share some logs I can give you pointers wrt what to look at.

**Comment by xf57212** [comment]
Sorry, this should have been closed out a while ago as we found the issue.

There was a conflict between Netflix Eureka and Docker Swarm both trying to load balance our services. Docker swarm seemed to have taken priority causing all instances to be registered as a single instance within Netflix Eureka. At the time of this ticket we were not aware Docker Swarm had any load balancing. We have since fixed the conflict and everything is working as intended now.
